### PR TITLE
chore(tests): add httproutematch builder utility to reduce parser tests

### PR DESCRIPTION
**Pull request by jrsmroz**
**What this PR does / why we need it**:

**Which issue this PR fixes**:

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Review by pmalek — APPROVED** [review]

File: internal/util/builder/httproutematch.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	return b.WithPathType(&pathPrefix, (*gatewayv1beta1.PathMatchType)(pointer.StringPtr(string(gatewayv1beta1.PathMatchPathPrefix))))
+}
+
+func (b *HTTPRouteMatchBuilder) WithPathRegex(pathRegexp string) *HTTPRouteMatchBuilder {
+	return b.WithPathType(&pathRegexp, (*gatewayv1beta1.PathMatchType)(pointer.StringPtr(string(gatewayv1beta1.PathMatchRegularExpression))))
+}
+
+func (b *HTTPRouteMatchBuilder) WithPathExact(pathRegexp string) *HTTPRouteMatchBuilder {
+	return b.WithPathType(&pathRegexp, (*gatewayv1beta1.PathMatchType)(pointer.StringPtr(string(gatewayv1beta1.PathMatchExact))))
+}
```
Comment (czeslavo) [inline internal/util/builder/httproutematch.go:36]: Would that make sense to do the casting magic in `WithPathType` to not duplicate it? 🤔

Comment (czeslavo) [reply internal/util/builder/httproutematch.go:36]: Ah, I see, `WithPathType` is public and it accepts `*gatewayv1beta1.PathMatchType`. Maybe a private helper `withPathType` accepting strings?

Comment (jrsmroz) [reply internal/util/builder/httproutematch.go:36]: It's public as both fields are actually optional

Comment (jrsmroz) [reply internal/util/builder/httproutematch.go]: @czeslavo ptal

**Review by czeslavo — APPROVED** [review]

**Pull request merged**